### DISCUSSION TITLE
Query on API Support for Manually Accepting Invitations

### DISCUSSION THREAD

**Discussion by kamigerami** [clerk/javascript#2117] [Support request - DO NOT OPEN HERE] [answered]
Hello,

I'm currently implementing a feature in our application that involves sending invitations to users via Clerk. I've encountered a specific use case and I'm seeking guidance on how to best handle it with Clerk's functionalities.

Use Case:
We are sending invitations to users to join a specific company within our application. However, we've noticed that if a user already has an account in clerk (but is not yet associated with the company in our db), the invitation remains in a 'pending' state even after they log in (through sign-in instead of sign-up). This happens because these users don't go through the typical invitation acceptance process, as they already have an account.

Question:
Is there an API endpoint or a method in Clerk that allows us to manually set the status of an invitation to 'accepted'? This would be particularly useful for instances where the user already has an existing account in our system and we want to streamline their experience by bypassing the standard invitation acceptance process.

Any guidance or alternative suggestions for handling this scenario would be greatly appreciated.

Thank you for your assistance.

Best regards,

**Comment by panteliselef** [comment]
Hey @kamigerami, just to confirm, are u referring to Organization Invitations ?
Do u have a custom flow for implementing the sign-in/up pages, do you take advantage of our prebuilt UI Components or do you use Account Portal ?

Are you sending invitations from our Backend API or through clerk-js ?

Can you provide more information regarding the following sentence ?
> This happens because these users don't go through the typical invitation acceptance process, as they already have an account.

**Reply by kamigerami** [reply]
Hello @panteliselef ! Thank you for the quick response.

Specifically, I'm using clerkClient.invitations.createInvitation({....}) from @clerk/nextjs/server.

Here's the scenario: We have existing users in Clerk who haven't been linked to a company in our database. When another user invites them, they're placed in a "pending" state. However, if these invited users try to sign up using the link sent to them, they're informed that an account already exists and are prompted to sign in instead. After signing in, their status remains "pending" rather than changing to "accepted."

We need a way to update the invitation status to "accepted" once these users log in. Is there a way to set the invitation status manually with the api?

We are not using the Organization Invitations.

**Reply by panteliselef** [reply]
Do you mind sharing if the users sign-in through Account portal, our UI components, or a custom flow of yours ?

**Reply by kamigerami** [reply]
They sign-in through the clerk/nextjs `<SignIn> / <SignUp>` components.

**Comment by gkats** [comment, marked as answer]
Hello @kamigerami :wave: . Thank you for providing more details for your use case.

The invitations feature is intended for inviting new users to your application. It's supposed to be helpful for targeting specific users or enabling invite-only registrations for your application. In general, you wouldn't want to invite an already existing user to your application.

I understand that your use case is a bit different though, so let's explore a few options. 

Would it be ok to [revoke the invitation](https://clerk.com/docs/reference/backend-api/tag/Invitations#operation/RevokeInvitation) when the user signs in? Revoked invitations don't count as pending. The revoked status might be appropriate after all; the user already exists so there's no point in inviting them again.

As an aside, have you checked out our [Organizations related features](https://clerk.com/docs/organizations/overview)? Sounds like what you call "company" in your application might fit the description of a Clerk organization. 

You can send [invitations for an organization](https://clerk.com/docs/reference/backend-api/tag/Organization-Invitations) to both existing and new users for your application. Whenever a user joins the organization, the invitation will be marked as "accepted".

Hope the above are helpful. Please let us know if you'd like further assistance.

**Reply by kamigerami** [reply]
Hello @gkats, yes I believe revoking is a possible fix.
We have not looked into the organizations features but will do that, it might be a better fit for us as you mentioned.

Thank you, I will look into it further  :)
